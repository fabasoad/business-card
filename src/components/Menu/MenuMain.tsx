import * as React from 'react'
import { Navbar, Nav } from 'react-bootstrap'
import { useTranslation } from 'react-i18next'

import LocaleDropDown from './LocaleDropDown'
import MenuItem from './MenuItem'

export default function MenuMain() {
  const { t } = useTranslation()

  const [expanded, setExpanded] = React.useState<boolean>(false)
  const [activeNavLink, setActiveNavLink] = React.useState<string>(document.location.hash)

  document.querySelectorAll('*').forEach((el) => {
    el.addEventListener('click', (e) => {
      const findParent = (el1: HTMLElement, el2: HTMLElement): HTMLElement => {
        if (el1 === el2) {
          return el1
        }
        if (el1.parentElement == null) {
          return null
        }
        return findParent(el1.parentElement, el2)
      }
      const element = e.target as HTMLElement
      const isNotNav = element.getAttribute('id') != 'nav'
      const isNotParentOfNav = findParent(document.getElementById('nav'), element) == null
      const isNavIsNotParent = findParent(element, document.getElementById('nav')) == null

      if (isNotNav && isNotParentOfNav && isNavIsNotParent) {
        setExpanded(false)
      }
    })
  })

  return (
    <div className="affix-top" id="nav">
      <Navbar expanded={expanded} onToggle={setExpanded} collapseOnSelect={true} expand="md" className="navbar-custom">
        <Navbar.Brand href="#header">{t('business-card-author-name')}</Navbar.Brand>
        <Navbar.Toggle className="navbar-dark" aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav activeKey={activeNavLink} className="mr-auto">
            <MenuItem setActiveNavLink={setActiveNavLink} name={'about'} />
            <MenuItem setActiveNavLink={setActiveNavLink} name={'skills'} />
            <MenuItem setActiveNavLink={setActiveNavLink} name={'experience'} />
            <MenuItem setActiveNavLink={setActiveNavLink} name={'education'} />
            <MenuItem setActiveNavLink={setActiveNavLink} name={'portfolio'} />
            <MenuItem setActiveNavLink={setActiveNavLink} name={'resume'} />
            <MenuItem setActiveNavLink={setActiveNavLink} name={'contact'} />
          </Nav>
          <Nav>
            <LocaleDropDown />
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </div>
  )
}
