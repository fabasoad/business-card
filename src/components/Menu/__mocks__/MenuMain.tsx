import * as React from 'react'

export default function MenuMain() {
  return (
    <div data-testid="MenuMain" />
  )
}
