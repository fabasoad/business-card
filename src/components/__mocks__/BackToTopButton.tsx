import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
