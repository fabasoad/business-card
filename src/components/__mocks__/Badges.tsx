import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
