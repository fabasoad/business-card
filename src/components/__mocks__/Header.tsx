import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
