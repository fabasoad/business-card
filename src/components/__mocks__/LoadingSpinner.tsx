import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
