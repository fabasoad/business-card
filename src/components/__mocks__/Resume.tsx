import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
