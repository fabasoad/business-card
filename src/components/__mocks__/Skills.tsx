import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
