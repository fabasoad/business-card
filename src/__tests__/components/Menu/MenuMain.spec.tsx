/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import MenuMain from '../../../components/Menu/MenuMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render MenuMain correctly', () => {
  const wrapper: ShallowWrapper = shallow(<MenuMain />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(1)
  expect(tMock).toHaveBeenCalledWith('business-card-author-name')
})
