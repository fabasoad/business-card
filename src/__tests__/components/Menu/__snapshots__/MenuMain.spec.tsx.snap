// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MenuMain correctly 1`] = `
<div
  className="affix-top"
  id="nav"
>
  <Navbar
    className="navbar-custom"
    collapseOnSelect={true}
    expand="md"
    expanded={false}
    onToggle={[Function]}
    variant="light"
  >
    <NavbarBrand
      href="#header"
    >
      business-card-author-name
    </NavbarBrand>
    <NavbarToggle
      aria-controls="basic-navbar-nav"
      className="navbar-dark"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="basic-navbar-nav"
    >
      <Nav
        activeKey=""
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <MenuItem
          name="about"
          setActiveNavLink={[Function]}
        />
        <MenuItem
          name="skills"
          setActiveNavLink={[Function]}
        />
        <MenuItem
          name="experience"
          setActiveNavLink={[Function]}
        />
        <MenuItem
          name="education"
          setActiveNavLink={[Function]}
        />
        <MenuItem
          name="portfolio"
          setActiveNavLink={[Function]}
        />
        <MenuItem
          name="resume"
          setActiveNavLink={[Function]}
        />
        <MenuItem
          name="contact"
          setActiveNavLink={[Function]}
        />
      </Nav>
      <Nav
        fill={false}
        justify={false}
      >
        <Connect(LocaleDropDown) />
      </Nav>
    </NavbarCollapse>
  </Navbar>
</div>
`;
